### DISCUSSION TITLE
Only congestive_heart_failure patients

### DISCUSSION THREAD

**Discussion by johagg17** [synthetichealth/synthea#1033] [General]
### Requested Feature

Hi, 

I am working on a project where I will use Synthea to generate patient records, for heart failure patients. 
I saw that you provide the 'congestive_heart_failure' module, but when I run the program as below, only a small portion of the patients is related to the HF, i.e. length of patient.csv is as specified 500k patients while conditions.csv only contains a few. 

run_synthea -p 500000 -a 18-90 -m congestive_heart_failure 

Why is this the case, and is there any way I can generate records such that all 500k patients are related to the congestive_heart_failure module? 

Thanks for now, and thanks for providing such a great tool as Synthea.

**Comment by jawalonoski** [comment]
One issue is likely to be your use of the M feature. Please see this page of discussion on the M feature: https://github.com/synthetichealth/synthea/wiki/The--M-Feature

If you want only CHF patients, I would suggest not using the M feature (omit that completely and run all the disease models together), and add a Keep Module.

See this page about Keep Modules: https://github.com/synthetichealth/synthea/wiki/Keep-Patients-Module

**Reply by johagg17** [reply]
Hi, 

Thank you, one final question,

I experience some issues when trying to run the -k command, and the program does not seem to find the keep module file. 

Error: java.io.FileNotFoundException: specified keep-patients file (keep_heart_failure.json) does not exist.
 
Where should I put this Keep Module file, same as the other modules (MyComputer\synthea\src\main\resources\modules)?

Thanks, once again.

**Reply by jawalonoski** [reply]
I tend to keep them directly in the root folder of the project (in the same level as the `run_synthea` script).

You don't want to add them to the other modules, because you don't want the program to consider them to be actual disease models.

So, if I have a file `keep.json` I just keep it in the root Synthea directory and run `./run_synthea -k keep.json` (with whatever other parameters you may also want).
